# Excellente resources papers for genomics



- #### The genome sequence of tetraploid sweet basil, ocimum basilicum L., provides tools for advanced genome editing and molecular breeding [DNA Research](https://academic.oup.com/dnaresearch/advance-article/doi/10.1093/dnares/dsaa027/6042144?login=true#supplementary-data)


- #### EXCLLENT INSPIRATION INDICUM GENOME | Good comparison example and graph with R | [Brassica napus](https://academic.oup.com/gigascience/article/9/12/giaa137/6034787) | [Cucumis sativus](https://academic.oup.com/gigascience/article/8/6/giz072/5520540?searchresult=1)

- #### [Phased diploid genome assemblies and pan-genomes provide insights into the genetic history of apple domestication](https://www.nature.com/articles/s41588-020-00723-9)

- #### [Excellent tutot for full annotation and downstream analysis | Chromosomal genome of Triplophysa bleekeri provides insights into its evolution and environmental adaptation](https://academic.oup.com/gigascience/article/9/11/giaa132/5999722)


- #### [Genome-enabled discovery of anthraquinone biosynthesis in Senna tora](https://www.nature.com/articles/s41467-020-19681-1)


- #### [A reference high-density genetic map of Theobroma grandiflorum (Willd. ex Spreng) and QTL detection for resistance to witches’ broom disease (Moniliophthora perniciosa)](https://link.springer.com/article/10.1007/s11295-020-01479-3)




- #### [Phylogenomics resolves the invasion history of Acacia auriculiformis in Florida](https://onlinelibrary.wiley.com/doi/full/10.1111/jbi.14013)


- #### [Nice graph heterozigosity](https://www.frontiersin.org/articles/10.3389/fgene.2020.00440/full)


- #### [Guide writting short format](https://www.biorxiv.org/content/10.1101/2020.04.18.042093v1.full.pdf)

- #### [Awesome tutorial before wgs assembly](https://www.grandomics.com/en/complete_genome_survey/)

- #### [Awesome glance about de nov assembly and comparative genomics](https://www.grandomics.com/en/animal_and_plant_genome_de_novo/)

- #### [Grandomics genomics company pub](https://www.grandomics.com/en/pubications/)


- #### [Zoonomia project](https://www.nature.com/articles/s41586-020-2876-6) | [Code availability](https://www.nature.com/articles/s41586-020-2876-6#code-availability) 

- #### Gigascience Data Note Genome sequence | [Callicarpa americana](https://academic.oup.com/gigascience/article/9/9/giaa093/5902285) | [Potato](https://academic.oup.com/gigascience/article/9/9/giaa100/5910251#207670453) | [Aquilaria sinensis](https://academic.oup.com/gigascience/article/9/3/giaa013/5771296) | [Arachis monticola](https://academic.oup.com/gigascience/article/7/6/giy066/5040258) | [Eriobotrya japonica](https://academic.oup.com/gigascience/article/9/3/giaa015/5788433) |[Portunus trituberculatus](https://academic.oup.com/gigascience/article/9/1/giz161/5697200) | [catfish](https://academic.oup.com/gigascience/article/7/11/giy120/5106933) | [Juglans regia](https://academic.oup.com/gigascience/article/9/5/giaa050/5841058) |  [Juglans sigillata](https://academic.oup.com/gigascience/article/9/2/giaa006/5758190) | [mustache toad](https://academic.oup.com/gigascience/article/8/9/giz114/5572531) | [Prunus](https://academic.oup.com/gigascience/article/9/12/giaa130/6029397?searchresult=1)

- #### [Molecular Plant | Collection: Plant Genomics](https://www.cell.com/molecular-plant/collections/plant-genomics)

- #### [High-Resolution Genome-wide Association Study Identifies Genomic Regions and Candidate Genes for Important Agronomic Traits in Wheat](https://www.cell.com/molecular-plant/fulltext/S1674-2052(20)30221-5)  | GWAS | Population genomics

- #### [The Genomes of the Allohexaploid Echinochloa crus-galli and Its Progenitors Provide Insights into Polyploidization-Driven Adaptation](https://www.cell.com/molecular-plant/fulltext/S1674-2052(20)30214-8) | Tracking polyploidy 

- #### [The Chromosome-level Reference genome Assembly for Panax notoginseng and Insights into Ginsenoside Biosynthesis](https://www.cell.com/plant-communications/fulltext/S2590-3462(20)30144-9)

- #### [A Chromosome-Level Genome Assembly of Garlic (Allium sativum) Provides Insights into Genome Evolution and Allicin Biosynthesis](https://www.cell.com/molecular-plant/fulltext/S1674-2052(20)30232-X)

- #### [Deciphering the high‐quality genome sequence of coriander that causes controversial feelings](https://onlinelibrary.wiley.com/doi/full/10.1111/pbi.13310) | [Coriander Genomics Database: a genomic, transcriptomic, and metabolic database for coriander](https://www.nature.com/articles/s41438-020-0261-0)| MCMCTREE | DATABASE


- #### [The Chromosome-Level Genome Sequence of the Autotetraploid Alfalfa and Resequencing of Core Germplasms Provide Genomic Resources for Alfalfa Research](https://www.cell.com/molecular-plant/fulltext/S1674-2052(20)30216-1) | colinearity | MCscan | Ka Ks | Reseq | SNP | GATK

- #### [Development and evaluation of high‐density Axiom®CicerSNP Array for high‐resolution genetic mapping and breeding applications in chickpea](https://onlinelibrary.wiley.com/doi/full/10.1111/pbi.12836)

- #### [Toward the sequence-based breeding in legumes in the post-genome sequencing era](https://link.springer.com/article/10.1007/s00122-018-3252-x)

- #### [Integrating genomics for chickpea improvement: achievements and opportunities](https://link.springer.com/article/10.1007/s00122-020-03584-2)


- #### 07 September 2020 | Article | [Generation of a chromosome-scale genome assembly of the insect-repellent terpenoid-producing Lamiaceae species, Callicarpa americana](https://academic.oup.com/gigascience/article/9/9/giaa093/5902285)

- #### 28 August 2020 | Article | [The genome of the cauliflower coral Pocillopora verrucosa](https://academic.oup.com/gbe/advance-article/doi/10.1093/gbe/evaa184/5898631)

- #### 1 August 2020 | Article | [Chromosome-scale genome assembly of sweet cherry (Prunus avium L.) cv. Tieton obtained using long-read and Hi-C sequencing](https://www.nature.com/articles/s41438-020-00343-8)

- #### 8 May 2020 | Article | [Chromosome-level assembly of the horseshoe crab genome provides insights into its genome evolution](https://www.nature.com/articles/s41467-020-16180-1)
- #### 29 July 2020 | Article | [Gradual polyploid genome evolution revealed by pan-genomic analysis of Brachypodium hybridum and its diploid progenitors](https://www.nature.com/articles/s41467-020-17302-5)
- #### 1 August 2020 | Article | [A high-quality sponge gourd (Luffa cylindrica) genome](https://www.nature.com/articles/s41438-020-00350-9)
- #### 10 August 2020 | Article | [A high-contiguity Brassica nigra genome localizes active centromeres and defines the ancestral Brassica genome](https://www.nature.com/articles/s41477-020-0735-y#Sec4)
- #### 29th January 2020 | Article | [High quality 3C de novo assembly and annotation of a multidrug resistant ST-111 Pseudomonas aeruginosa genome: Benchmark of hybrid and non-hybrid assemblers](https://www.nature.com/articles/s41598-020-58319-6)
- #### 20 July 2020 | Review | [Plant pan-genomes are the new reference](https://www.nature.com/articles/s41477-020-0733-0)

- #### 01 October 2020 |Article | [From landrace to modern hybrid broccoli: the genomic and morphological domestication syndrome within a diverse B. oleracea collection](https://www.nature.com/articles/s41438-020-00375-0) | [Code available](https://github.com/zacharystansell/B_oleracea_diversity_panel)

- #### [Example linkgae map construction](https://www.mdpi.com/2223-7747/9/11/1476)

# Selection pressure analysis

- #### [Selection signatures of Fuzhong Buffalo based on whole-genome sequences](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-07095-8)

# Pangenome


- #### [definition](https://www.grandomics.com/en/pan-genome/)


- #### Apple pan-genome | 02 Noember 2020 | Domestication | Assembly improvement | [Phased diploid genome assemblies and pan-genomes provide insights into the genetic history of apple domestication](https://www.nature.com/articles/s41588-020-00723-9) | [Code availability](https://github.com/XuepengSun/apple_diploid_genomes)

- #### PacBIO a dit "Sequencing multiple individuals is the best way to understand genomic variation in a species or across closely related species." [photo](https://programs.pacificbiosciences.com/l/1652/2020-09-18/426l8j)

- #### [Eight high-quality genomes reveal pan-genome architecture and ecotype differentiation of Brassica napus](https://www.nature.com/articles/s41477-019-0577-7)

- #### [Discovery and population genomics of structural variation in a songbird genus](https://www.nature.com/articles/s41467-020-17195-4)

- #### [Pan-Genome of Wild and Cultivated Soybeans](https://www.sciencedirect.com/science/article/abs/pii/S0092867420306188)

- ##### [Six reference-quality genomes reveal evolution of bat adaptations](https://www.nature.com/articles/s41586-020-2486-3)

- #### [Sunflower pan-genome analysis shows that hybridization altered gene content and disease resistance](https://www.nature.com/articles/s41477-018-0329-0)

- #### [Beyond a Single Reference Genome – The Advantages of Sequencing Multiple Individuals](https://youtu.be/mfldTjg1EqI)
- #### [Rice](https://www.nature.com/articles/s42003-020-0890-8#Sec8) | [Rice wild](https://link.springer.com/article/10.1007/s11427-020-1738-x)
- #### [Plant pan-genomes are the new reference](https://www.nature.com/articles/s41477-020-0733-0)
- #### [Super-Pangenome by Integrating the Wild Side of a Species for Accelerated Crop Improvement](https://www.cell.com/trends/plant-science/fulltext/S1360-1385(19)30281-X)

# Root sesame

- #### Comparative Analysis of Root Transcriptome Profiles of Sesame (Sesamum indicum L.) in Response to Osmotic Stress](https://link.springer.com/article/10.1007/s00344-020-10230-0)


# Resequencing wild and cultivared


- ####  [2010 | Soybean 31](https://www.nature.com/articles/ng.715)
- ####  [2013 | Soybean | Korea](https://academic.oup.com/dnaresearch/article/21/2/153/403337)
- ####  [2015 | Soybean 302](https://www.nature.com/articles/nbt.3096)
- ####  [2014 | pepper](https://www.pnas.org/content/111/14/5135)
- ####  [2019 | Eggplant](https://www.frontiersin.org/articles/10.3389/fpls.2019.01220/full)
- ####  [2020 | Jujube](https://onlinelibrary.wiley.com/doi/10.1111/pbi.13480)
- ####  [2020 | Lotus](https://onlinelibrary.wiley.com/doi/abs/10.1111/tpj.15029)
- ####  [2020 | watermelon | 414](https://www.nature.com/articles/s41588-019-0518-4)
- ####  [2020 | Soybean](https://www.sciencedirect.com/science/article/pii/S0092867420306188?casa_token=v9ShlMGm3gQAAAAA:Sx5ybFWtapUYcU99V3aUWCsaaUWWqwvGmHlDdgud6q2yYWllCEV8oORQTueThOkpDgW09AAvoQ)
- ####  [2020 | Alfafa](https://www.sciencedirect.com/science/article/pii/S1674205220302161?casa_token=ocSWUtmqUeYAAAAA:ZUbB89_FygJ3F2EgClSg2EY6rVJD2A0HKAWVn9K9qtCSIEgBMoIZZSzv-CEX8Pu3jSX8HiSeJw)
- ####  [2020 | Tea | 81](https://www.sciencedirect.com/science/article/pii/S1674205220301349)
- ####  [2020 | Rice | 1143](https://www.nature.com/articles/s41467-020-18608-0)
- ####  [2020 | Common Bean | 683](https://www.nature.com/articles/s41588-019-0546-0)
- #### [2020 | Pepper | 4 | Italia](https://www.nature.com/articles/s41598-020-66053-2)

# GWAS AND GENETIC DIVERSITY

- #### [A Genomics Resource for Genetics, Physiology, and Breeding of West African Sorghum](https://www.biorxiv.org/content/10.1101/2020.06.03.132217v1.full)


- #### 02 November 2020 [Genome-wide association for β-glucan content, population structure, and linkage disequilibrium in elite oat germplasm adapted to subtropical environments](https://link.springer.com/article/10.1007/s11032-020-01182-0)

- #### 17 September 2020 | [A comprehensive phenotypic and genomic characterization of Ethiopian sorghum germplasm defines core collection and reveals rich genetic potential in adaptive traits](https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/tpg2.20055)
- #### 18 September 2020 | [Genomic analyses of diverse wild and cultivated accessions provide insights into the evolutionary history of jujube](https://onlinelibrary.wiley.com/doi/10.1111/pbi.13480)

# Polyploid

- #### [The relation of meiotic behaviour to hybridity, polyploidy and apomixis in the Ranunculus auricomus complex (Ranunculaceae)](https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-020-02654-3)

- #### [Polyploidy and high environmental tolerance increase the invasive success of plants](https://link.springer.com/article/10.1007/s10265-020-01236-6)


# Books

- #### [Computationnal genomics with R](https://compgenomr.github.io/book/)


# Snakemake


- #### [Install snakemake using conda](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html)
- #### [Introduction to snakemake](https://blog.liang2.tw/2017Talk-Snakemake/#cover) | [A video](https://www.youtube.com/watch?v=aZ0QlqtR2KM)
- #### [Run snakemake locally](https://jrderuiter.github.io/snakemake-rnaseq/usage.html)
- #### [CulebrONT: Using data from long reads obtained by Oxford Nanopore Technologies sequencing makes genome assembly easier](https://culebront-pipeline.readthedocs.io/en/latest/)
- #### [VIPER: Visualization Pipeline for RNA-seq, a Snakemake workflow for efficient and complete RNA-seq analysis](https://bitbucket.org/cfce/viper/src/master/#RunningViper) 

- ##### [snakemake rna seq star deseq2](https://github.com/snakemake-workflows/rna-seq-star-deseq2)

- #### [RASflow: an RNA-Seq analysis workflow with Snakemake](https://github.com/zhxiaokang/RASflow)

- #### [Snake make Hista Strintie Deseq](https://github.com/KoesGroup/Snakemake_hisat-DESeq)

- #### [Variant calling example with snakemake](https://snakemake.readthedocs.io/en/v5.25.0/tutorial/basics.html)

- #### [snakemake-workflows dna-seq-gatk-variant-calling](https://github.com/snakemake-workflows/dna-seq-gatk-variant-calling/blob/master/Snakefile)






# Genomics Tools


- #### Wonderfull tool for plastid representation [website](https://irscope.shinyapps.io/Chloroplot/) | [paper](https://www.readcube.com/articles/10.3389/fgene.2020.576124)

- #### [LongQC: A Quality Control Tool for Third Generation Sequencing Long Read Data](https://www.g3journal.org/content/ggg/10/4/1193.full.pdf)
- #### [D-GENIES – for Dot plot large Genomes in an Interactive, Efficient and Simple way – is an online tool designed to compare two genomes. It supports large genome and you can interact with the dot plot to improve the visualisation.](http://dgenies.toulouse.inra.fr/)

- #### [Genome Assembly with Minimap2 and Miniasm with pacbio data](https://github.com/lh3/miniasm)
- #### [Genome Assembly with Minimap2 and Miniasm with nanopore data](https://timkahlke.github.io/LongRead_tutorials/ASS_M.html)

- #### [Nanopore explanation reads trimming and so on:  Telomere length de novo assembly of all 7 chromosomes and mitogenome
sequencing of the model entomopathogenic fungus, Metarhizium brunneum,
by means of a novel assembly pipeline
](https://assets.researchsquare.com/files/rs-60098/v2/e4035f37-970c-481e-8d23-05b18f9783b6.pdf)

- #### [IJV Tool](https://wikis.utexas.edu/display/bioiteam/Integrative+Genomics+Viewer+%28IGV%29+tutorial)

- #### [Assembly Graph Browser: interactive visualization of assembly graphs](https://pubmed.ncbi.nlm.nih.gov/30715194/)

- #### [Visualising Assembly Graphs](https://towardsdatascience.com/visualising-assembly-graphs-fb631f46bbd1)

- #### [Excellent tutorials for long reads assembly and so on](https://yiweiniu.github.io/blog/)
- #### [Accessing data from NCBI with EDirect](https://astrobiomike.github.io/unix/ncbi_eutils)

- #### [Webinar: GWA of a wheat transcription factor family: the power of bioinformatics resources](https://youtu.be/efbph7f2jXU)

- #### [Hi-fi assembly yutorial](https://youtu.be/5tE8zYDpC8c)

- #### [Long reads assembly tutorial with a focus on nanopore data](https://timkahlke.github.io/LongRead_tutorials/ASS.html)
- #### [How to read a phylogenetic tree](https://youtu.be/LgZ3UbPzC6c)

- #### [Submitting a haploid assembly: submitting WGS contigs only](https://www.ncbi.nlm.nih.gov/assembly/docs/submission/#ex1)

- #### [Genome colinearity analysis tool: MCScan](https://github.com/tanghaibao/mcscan)
- #### [Genome colinearity analysis tool: MCScanX](https://github.com/wyp1125/MCScanX)
- #### [Genome colinearity visualization tool Synvisio](https://github.com/kiranbandi/synvisio)

- #### [Shark: fishing relevant reads in an RNA-Seq sample](https://doi.org/10.1093/bioinformatics/btaa779)
- #### [TBtools](https://github.com/CJ-Chen/TBtools) | [Paper](https://www.sciencedirect.com/science/article/pii/S1674205220301878)
- #### [Ka ks calculation with TBtools](https://www.youtube.com/watch?v=-KjSdr-hmk4)
- #### [Exon-intron identification](https://www.youtube.com/watch?v=60V0WscNEQE)
- #### [Syntheny analysis in TBtools](https://www.youtube.com/watch?v=v-t0z0UrHqQ)
- #### [ka ks online](http://services.cbu.uib.no/tools/kaks)
- #### [Best Practices/Softwares To Calculate Ka/Ks Ratio](https://www.biostars.org/p/5817/)
- #### [Bayesian Molecular Clock Dating Using Genome-Scale Datasets](https://link.springer.com/protocol/10.1007/978-1-4939-9074-0_10)
- #### [MCMCtreeR: functions to prepare MCMCtree analyses and visualize posterior ages on trees](https://academic.oup.com/bioinformatics/article-abstract/35/24/5321/5530964?redirectedFrom=fulltext)  |  [MCMtreeR](https://cran.r-project.org/web/packages/MCMCtreeR/vignettes/MCMCtree_plot.html)  | [MCMtreeR vignette](https://cran.r-project.org/web/packages/MCMCtreeR/vignettes/MCMCtree.html)



- #### RepeatModeler instalation 

[official](https://github.com/Dfam-consortium/RepeatModeler)

[1 blog](http://kazumaxneo.hatenablog.com/entry/2020/07/03/073000) | [2 official](http://www.repeatmasker.org/RepeatModeler/) | [3 incodom](http://www.incodom.kr/RepeatModeler)  | [4 docker](https://hub.docker.com/r/pakorhon/repeatmodeler)  | [5 installation guide](http://www.hpc.lsu.edu/docs/faq/installation-details.php)  | [6](https://taoyan.netlify.app/post/2020-03-02.edta-%E8%BD%AC%E5%BA%A7%E5%AD%90%E6%B3%A8%E9%87%8A/)   | [7](https://www.jianshu.com/p/a8c9e875b17f)  |  [8](https://www.douban.com/note/777662891/)   |  [9](https://www.programmersought.com/article/25101639140/)  |  



[un site par haserd tutorial interessant](http://www.wuchangsong.com/?cat=4)

[edta](https://blog.csdn.net/u012110870/article/details/103903879)


[astuce](https://www.qdxingtu.com/key/Error%20running%20repeatModeler.html)


```

astuce... 

The only thing that worked for me was to install all dependencies in a conda environment and installing RepeatModeler 2.0.1 and RepeatMasker 4.1.0 downloaded from GitHub separately (not with conda) and linking all dependencies of the two configure scripts to the conda environment.
```




# Singularity

[1](https://www.sdsc.edu/education_and_training/tutorials1/singularity_old.html)


# R shiny

- #### [Shiny Apps: Development and Deployment](https://www.mzes.uni-mannheim.de/socialsciencedatalab/article/shiny-apps/)
- #### [Shiny Apps: Development and Deployment Video](https://www.youtube.com/watch?v=QT3WUQu99pM)
- #### [R Recipe: Shiny Tables from Excel Templates](https://ljupcho.com/blog/shiny-excel-table-templates)



# R

- #### Heatmap 


```

library(pheatmap); library(gplots); library(ggplot2); library(colorspace); library(wesanderson)

tpm = delim("Senna_tora_CHS-L.txt", header=T, row.names='gene')
bk = unique(c(seq(0, 0, length=10), 0, seq(0, 13, length=100)))
colors = colorRampPalette(c("white", "red"))(100)
pheatmap(tpm, 
         color = colors,
         breaks = bk,
         cluster_cols=F,
         cluster_rows=F,
         fontsize_col=9,
         fontsize_row=7,
         cellwidth = 260/ncol(tpm), 
         cellheight = 200/nrow(tpm), 
         border_color = "grey",
         show_rownames=T,
         gaps_col = c(1,2,3,4,5,6),
         labels_row = rownames(tpm),
         main=paste(c(nrow(tpm), " CHS-L genes"),collapse=""))
         
 
  ```
 
 
 - #### Ks plot
 
 
 ```
 
 library(ggplot2); library(ggrepel); library(gridExtra)
args = commandArgs(trailingOnly=TRUE)

library(limma)
Cats = c()
KaKs = data.frame()
for (i in args){
  Data = data.frame(category = strsplit2(i, '\\.')[1,1], read.table(i, sep = '', header = T, quote = '', fill = T))
  KaKs = rbind(KaKs, Data)
  Label = strsplit2(i, '\\/')
  Cats = c(Cats, strsplit2(Label[1, ncol(Label)], '\\.')[1,1])
}

KaKs$category = factor(KaKs$category, levels = Cats)
NumOrg = length(Cats)


Ks = ggplot(KaKs) + 
  geom_density(aes(Ks, fill = factor(category)), alpha = 0.3, colour = NA) + 
  scale_fill_manual(values = rainbow(NumOrg)) +
  scale_x_continuous(expand = c(0,0), limits = c(0, 4)) + 
  ggtitle('Ks') +  
  theme_bw(base_size = 6) +
  xlab('Ks') +
  theme(legend.position = 'bottom',legend.title = element_blank(),plot.title = element_text(vjust = 0.5, hjust = 0.5, size = 20))

pdf(paste(c('./Plot_Ks.', Cats, '.pdf'), collapse = ''))
grid.arrange(Ks)
dev.off()

```



`

To calculate the synonymous-substitution Ks values, we selected the orthologous gene pairs between species and the paralogous pairs within a species from the orthology analysis. The selected proteins were further subjected to multiple-sequence alignment with MAFFT v7.305b93 and corrected with Gblocks v0.91b94. The corresponding genomic regions of conserved proteins, which were observed from the corrected multiple alignments, were subjected to Ks calculation using ParaAT v2.0 (ref. 98) with the Yang–Nielsen approach implemented in PAML99. The Ks distribution plot (Supplementary Fig. 27) was drawn using in-house Python and R scripts.

`





[kaks-calculator](https://code.google.com/archive/p/kaks-calculator/downloads)




         


- #### [R for biologist](http://omgenomics.com/plotting-in-r-for-biologists/)

- color palette   [inspiration](https://www.google.co.kr/imgres?imgurl=https%3A%2F%2Fplastid.readthedocs.io%2Fen%2Flatest%2F_images%2Fstackedbar.png&imgrefurl=https%3A%2F%2Fplastid.readthedocs.io%2Fen%2Flatest%2Fexamples%2Fz_plotting.html&tbnid=qxkqTSGPS4TqXM&vet=12ahUKEwj6p6Od6dntAhVD5ZQKHdO2C5gQMyhUegQIARBx..i&docid=Zi5Bizo5SpBWDM&w=1675&h=1324&q=genome%20plot%20python&hl=fr&ved=2ahUKEwj6p6Od6dntAhVD5ZQKHdO2C5gQMyhUegQIARBx) | [awespme](https://plastid.readthedocs.io/en/latest/examples/z_plotting.html)

- ### Principal Component Analysis | [STHDA Tuto](http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/112-pca-principal-component-analysis-essentials/) | [Visualization](http://www.sthda.com/english/wiki/fviz-pca-quick-principal-component-analysis-data-visualization-r-software-and-data-mining) | [factoextra tuto by Kassambara](https://rpkgs.datanovia.com/factoextra/reference/fviz_pca.html)

- #### [Tidytuesdays Tutorial](https://github.com/jkaupp/tidytuesdays)


- #### [A ggplot2 Tutorial for Beautiful Plotting in R](https://cedricscherer.netlify.app/2019/08/05/a-ggplot2-tutorial-for-beautiful-plotting-in-r/)

- #### [ggplot2 teaching material : Collection of teaching materials for R](https://github.com/Z3tt/ggplot-courses) 
- #### [ggplot2-tutorial](https://github.com/jennybc/ggplot2-tutorial)


- #### [super-and-sub-script-labels.R](https://gist.github.com/benmarwick/5cfeda1d20ff9ab3231a5de8fac36213)
- #### [polyshing a ggplot2 graph](https://ggplot2-book.org/polishing.html)


- #### [Remove 'a' from legend when using aesthetics and geom_text: Reponse 9](https://stackoverflow.com/questions/18337653/remove-a-from-legend-when-using-aesthetics-and-geom-text)


- #### [Find easily color hexa code](https://www.colorhexa.com/) | [website2](https://www.color-hex.com/)
- #### [Showing data values on stacked bar chart in ggplot2](https://intellipaat.com/community/12428/showing-data-values-on-stacked-bar-chart-in-ggplot2)


- #### [flextable](https://davidgohel.github.io/flextable/articles/overview.html)

- #### [y-axis on the right side · Len Kiefer](http://lenkiefer.com/2020/11/20/y-axis-on-the-right-side/)

- ### [acm](http://www.sthda.com/french/articles/38-methodes-des-composantes-principales-dans-r-guide-pratique/75-acm-analyse-des-correspondances-multiples-avec-r-l-essentiel/)

- #### [How to export high quality images in R](https://aebou.rbind.io/posts/2020/11/how-to-export-high-quality-images-in-r/)

- #### [Comment exporter des images de haute qualité avec R](https://aebou.rbind.io/fr/posts/2020/11/comment-exporter-des-images-de-haute-qualit%C3%A9-avec-r/)

- #### [New Version of Patchwork is Here: Inset a plot inside a plot](https://cmdlinetips.com/2020/11/insetting-with-patchwork/) [tuto2](https://www.data-imaginist.com/2020/insetting-a-new-patchwork-version/) | [tuto3](https://www.r-bloggers.com/2020/11/insetting-a-new-patchwork-version/) 

- #### [tips-to-combine-multiple-ggplots-using-patchwork](https://cmdlinetips.com/2020/01/tips-to-combine-multiple-ggplots-using-patchwork/)

- #### [Grouped Stacked bar plot with ggplot](https://www.r-graph-gallery.com/48-grouped-barplot-with-ggplot2.html)  | [Dealing with colors in ggplot2](https://www.r-graph-gallery.com/ggplot2-color.html) | [Paletteer package](https://github.com/EmilHvitfeldt/r-color-palettes)

- #### [Supprimer la legende](https://www.datanovia.com/en/fr/blog/gpplot-comment-supprimer-la-legende/)

- #### [Label bar chart ](https://thomasadventure.blog/posts/labels-ggplot2-bar-chart/)
- #### [inspi charles bodet ggplot2](https://www.charlesbordet.com/en/make-beautiful-charts-ggplot2/#)
- #### [BBC](https://bbc.github.io/rcookbook/)  [Exampple](http://www.interhacktives.com/2020/03/09/how-to-create-bbc-style-graphics-with-r-a-bbplot-tutorial/)
- #### [R graph galery ggplot2](https://www.r-graph-gallery.com/ggplot2-package.html)
- #### [Data to viz](https://www.data-to-viz.com/)

- #### [A ggplot2 Tutorial for Beautiful Plotting in R](https://cedricscherer.netlify.app/2019/08/05/a-ggplot2-tutorial-for-beautiful-plotting-in-r/)
- #### [Tidytuesday 2020](https://github.com/jwatzek/tidytuesday)

- #### inspi geom area en haut (plant height par example) et barre pourcentage en bas (pour le nombre de feuille vert et jaune et mort) , vrainspiration [ici](https://twitter.com/watzoever/status/1199508300959752192?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1199508300959752192%7Ctwgr%5Eshare_3%2Ccontainerclick_0&ref_url=https%3A%2F%2Fcedricscherer.netlify.app%2F2019%2F12%2F30%2Fbest-tidytuesday-2019%2F) ... le code dur github est [ici](https://github.com/jwatzek/tidytuesday)

- #### [inspi](https://cedricscherer.netlify.app/2020/01/09/australian-bushfires-comparison-europe-world/)

- #### [Inspi](https://frontpagedata.com/examples/)
- #### [axis label with ggplot2](http://environmentalcomputing.net/plotting-with-ggplot-adding-titles-and-axis-names/#:~:text=To%20alter%20the%20labels%20on,line%20of%20basic%20ggplot%20code.&text=Note%3A%20You%20can%20also%20use,which%20is%20equivalent%20to%20ggtitle%20.)
- ##### [annotation ggplot2](https://viz-ggplot2.rsquaredacademy.com/textann.html)
- #### [Ajouter les pvalues sur un ggplot, manuellement](https://statistique-et-logiciel-r.com/ajouter-pvalues-sur-ggplot/)

- #### [order x axis following a recise axix](https://www.r-graph-gallery.com/267-reorder-a-variable-in-ggplot2.html)

- #### [Modifier les axes color face et autres](http://www.sthda.com/french/wiki/ggplot2-graduation-des-axes-guide-pour-personnaliser-les-etiquettes-des-graduations-logiciel-r-et-visualisation-de-donnees)


- #### [barplot with error bar](https://www.r-graph-gallery.com/4-barplot-with-error-bar.html) or [here](http://www.sthda.com/french/wiki/ggplot2-barres-d-erreur-guide-de-demarrage-rapide-logiciel-r-et-visualisation-de-donnees)
- #### [Chicklet Stacked Bar Chart](https://www.mikelee.co/posts/2020-02-08-recreate-fivethirtyeight-chicklet-stacked-bar-chart-in-ggplot2/)

- #### [Distill for R Markdown: Creating a Website](https://rstudio.github.io/distill/website.html)

- #### [A Maize Color Palette Generator](https://github.com/AndiKur4/MaizePal)

- #### [Exporting editable ggplot graphics to PowerPoint with officer and purrr](https://www.pipinghotdata.com/posts/2020-09-22-exporting-editable-ggplot-graphics-to-powerpoint-with-officer-and-purrr/)
- #### [Learning Statistics with R](https://learningstatisticswithr.com/)
- #### [Becoming An rvest Magician](https://zachbogart.com/post/rvest-wizardry/)
- #### [How to zoom in on a plot in R](https://datavizpyr.com/how-to-zoom-in-on-a-plot-in-r/)

- #### [Top 50 Rggplot2 visualization items](http://r-statistics.co/Top50-Ggplot2-Visualizations-MasterList-R-Code.html)

- #### [Rboot Camp: A free online R course](https://r-bootcamp.netlify.app/)
- #### [The best books on Computer Science for Data Scientists recommended by Hadley Wickham](https://fivebooks.com/best-books/computer-science-data-science-hadley-wickham/)
- #### [flextable R package](https://davidgohel.github.io/flextable/)
- #### [PLOTTING SIGNIFICANCE ON THE TOP A BAR CHART OPTION1](https://nakedstatistics.wordpress.com/2017/05/11/plotting-significance/)
- #### [PLOTTING SIGNIFICANCE ON THE TOP A BAR CHART OPTION2](http://131.111.177.41/statistics/R/graphs2.html)
- #### [T TEST](http://www.sthda.com/english/wiki/unpaired-two-samples-t-test-in-r)
- #### [R studio data sheet](https://rstudio.cloud/learn/cheat-sheets)
- #### [R Cookbook](http://www.cookbook-r.com/)


# R Map

- #### [ggplot2 official guide for map](https://ggplot2-book.org/maps.html)


- #### [How to get terrain data and draw spatial maps in R using ggplot2? | StatswithR | Arnab Hazra](https://youtu.be/tX8RaTp7jL8)

- #### [Choropleth maps with geom_sf() | Professional dataviz with ggplot2 | R](https://youtu.be/nlcZcWJiyX8)

- #### [Raster maps with geom_raster() | Professional dataviz with ggplot2 | R](https://youtu.be/NQZNpyEgVss)

- #### [Create a map of a country with the neighbouring countries and an inset map](https://aebou.rbind.io/posts/2020/11/create-a-map-of-a-country-with-the-neighbouring-countries-and-an-inset-map/)

- #### [Interactive map with R](https://thinkr.fr/cartographie-interactive-avec-r-la-suite/)
- #### [Interactive map with R and Leaflet](https://thinkr.fr/cartographie-interactive-comment-visualiser-mes-donnees-spatiales-de-maniere-dynamique-avec-leaflet/)
- #### [Geocomputation with R](https://geocompr.robinlovelace.net/)
- #### [R as GIS, part 1: vector](https://datascienceplus.com/r-as-gis-part-1-vector/)

# R Markdown

- #### [Remedy provides addins to facilitate writing in markdown with RStudio](https://github.com/ThinkR-open/remedy) 


# Genomics


- #### [tutorial phylogeny](http://www.metagenomics.wiki/tools/phylogenetic-tree/construction)   [tuto](http://www.metagenomics.wiki/tools/phylogenetic-tree/alignment)

- #### [great example of comparative genomics of plants ...tu dois suive cet example pour sesamum alatum prevu pour gigascience  Usage de orthofinder avec option blast....phylogenetic and divergence time estimation wonderfull tutorial](https://www.frontiersin.org/articles/10.3389/fgene.2019.01211/full)

- #### [Great inspiration Murigneux Valentine](https://www.biorxiv.org/content/10.1101/2020.03.16.992933v1) | [Video](https://youtu.be/FuyMgjROOIk)

- #### [Code genome kiwifruit](https://www.protocols.io/view/chromosome-scale-genome-assembly-of-kiwifruit-acti-vgse3we)

- #### [Great article on why consider more than one reference genome](https://www.pacb.com/blog/sequencing101pangenome/)

- ### [SIB_LongReadsAsmWorkshop_Lausanne18](https://github.com/aechchiki/SIB_LongReadsAsmWorkshop_Lausanne18/wiki)

- #### [Bioinformatics Tutotial 2019](https://angus.readthedocs.io/en/2019/toc.html)

- #### [Computational genomics tutorial 2019](https://genomics.sschmeier.com/)

- #### [Understanding, Curating, and Analyzing your Diploid Genome Assembly](https://pb-falcon.readthedocs.io/en/latest/_downloads/Kingan_DiploidGenome_ECUGM2017_BFX.pdf)

- #### [Exercise: Long Read Assembly](https://scilifelab.github.io/courses/assembly/2017-11-15/exercises/Exercise_Long_Read_Assembly.html)

- #### [Long-reads assembly lecture](https://scilifelab.github.io/courses/assembly/2017-11-15/lectures/Long_read_Assembly.pdf)


- #### [Reproducible research in Genomics](https://nbis-reproducible-research.readthedocs.io/en/course_1803/)

- #### [Conda ORF finder](https://gist.github.com/mkweskin/30e3bd57534868cccd700f4181f07103)




# Transcriptomics

- #### [Maker Pipeline [2011 - v2.31.10 (May 4, 2018)](https://m.blog.naver.com/PostView.nhn?blogId=naturelove87&logNo=221491035712&proxyReferer=https:%2F%2Fwww.google.com%2F) | [MAKER](http://www.incodom.kr/MAKER) | [Methods](https://bioinformaticaupf.crg.eu/20132014/projectes1314/1C/Materials_and_methods.html)

- #### [Tuto for Tuxedo pipeline 1](https://colauttilab.github.io/NGS/TuxedoTutorial.html)

- ####  [Tuto for Tuxedo pipeline](https://github.com/trinityrnaseq/RNASeq_Trinity_Tuxedo_Workshop/wiki/Tuxedo-Genome-Guided-Transcriptome-Assembly-Workshop)

- ### [good tuto for deseq code and edger code](https://github.com/Yedomon/RNA-seq-tutorial-for-gene-differential-expression-analysis)

- #### [good tuto DESEQ](https://genviz.org/module-04-expression/0004/02/01/DifferentialExpression/)

- #### [Genomic Data Visualization and Interpretation](https://genviz.org/course/)

- #### [Excellent paper on RNA DEG spipeline](https://www.nature.com/articles/s41598-019-43421-1#Sec9)

- #### Deseq tuto [1](http://evomics.org/wp-content/uploads/2018/10/RNASeqWorkFlow-1.html) [2](https://rpubs.com/otienolwanga/problem_solving) [3](https://rpubs.com/mwken314/deseq2)

- #### HISAT2 HTseq DeSeq2 Rna tuto [1](http://www.cbs.dtu.dk/courses/27626/Exercises/rnaseq.php) | [2](https://pzweuj.github.io/2018/07/12/rna-seq-3.html) | [3 Hisat2 and featcount](https://bioinformaticsworkbook.org/dataAnalysis/RNA-Seq/RNA-SeqIntro/RNAseq-using-a-genome.html#gsc.tab=0) and [Deseq2](https://bioinformaticsworkbook.org/dataAnalysis/RNA-Seq/RNA-SeqIntro/Differential-Expression-Analysis.html#gsc.tab=0)


- #### [featureCounts and important note for aligner relative to short reads] (https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/05_counting_reads.html)

- #### [RNAseq workflow](https://github.com/Yedomon/RNAseq-workflow)

- #### [How to perform KEGG pathway analysis in R?](https://www.biostars.org/p/274514/) [Resources1](http://yulab-smu.top/clusterProfiler-book/chapter12.html) [Resources2](https://www.biostars.org/p/327039/)


- #### [Exact Tests and Plots with edgeR – Basic Differential Expression Analysis](https://morphoscape.wordpress.com/2020/09/01/exact-tests-and-plots-with-edger-basic-differential-expression-analysis/)

- #### [RNASeq lab](https://scilifelab.github.io/courses/ngsintro/1911/labs/rnaseq/lab.html)

- #### [Genomics and transcriptomics tutorials](https://scilifelab.github.io/courses/)

- #### [Annotation explained](https://www.biostars.org/p/181286/)

- #### [Fungal annotation script](https://github.com/Yedomon/Fungal_genome_annotation)

- #### [Annotation and comparative genomics script](https://github.com/Yedomon/AMPRIL-genomes)

- #### [RNA seq analysis](https://decodebiology.github.io/bioinfotutorials/rnaseq_tutorial.html)
- #### [RNA seq tuto](https://gif.biotech.iastate.edu/rnaseq-analysis-walk-through)

- #### [MA PLOT](https://www.biostars.org/p/457769/)

- #### [Complex heat map](https://github.com/kevinblighe/E-MTAB-6141)

- #### [A guide to elegant tiled heatmaps in R](https://www.royfrancis.com/a-guide-to-elegant-tiled-heatmaps-in-r-2019/)






# Chloroplast




- #### Comparative chloroplast genome analysis



[Evolutionary dynamics of the chloroplast genome sequences of six Colobanthus species](https://www.nature.com/articles/s41598-020-68563-5#Sec11)




1 Syntheny


>  Genome synteny analysis of the eight Colobanthus plastomes (six genomes reported in this paper, and C. quitensis and C. apetalus that were previously characterized and deposited in NCBI25,26) was performed with the use of MAUVE v.1.1.1


2 Ka and Ks



 >  The evolutionary rate of the plastome genes identified in all Colobanthus species (C. acicularis, C. affinis, C. apetalus, C lycopodioides, C. nivicola, C. pulvinatus, C. quitensis and C. subulatus) was analyzed. A total of 78 genes were selected to estimate the ratio of non-synonymous (Ka) to synonymous (Ks) substitutions. Colobanthus quitensis was the reference species. These genes were extracted and aligned separately using MAFFT v7.310. The values of Ka and Ks in the shared genes were calculated in DnaSP v.6.10.04. Genes with non-applicable (NA) Ka/Ks ratios were changed to zero.


3 nucleotide diversity


>  The sequences were aligned in MAFFT v.7.31096 to perform sliding window analysis and evaluate nucleotide diversity (π) in cp genomes using DnaSP v.6.10.0497. The step size was set to 50 base pairs, and window length was set to 800 base pairs.



4 Potential RNA editing sites



> Potential RNA editing sites in the protein-coding genes of chloroplast genomes were predicted using the Predictive RNA Editor for Plants (PREP) suite98. The cutoff value for the analyzed Colobanthus species was set at 0.8, and 34 out of the 35 reference genes in PREP were used. rpl23 was not included in the analysis because it was not identified within the chloroplast genomes of the studied Colobanthus species. Two previously sequenced cp genomes of C. quitensis and C. apetalus25,26 were also included in this analysis.



5 Phylogenetic analysis




>  The available (24) complete chloroplast genomes representing Caryophyllaceae lineages and the cp genome of Arabidopsis thaliana as an outgroup were downloaded from the NCBI database to investigate the phylogenetic relationships among the studied representatives of the genus Colobanthus and the genera in the family Caryophyllaceae. The cp genomes used in phylogenetic analyses are presented in Table 3. The sequences of 73 shared protein coding genes were extracted using custom R script, and they were aligned in MAFFT v.7.310. Finally, 73 concatenated protein-coding gene sequences where used for phylogeny reconstruction by Bayesian Inference (BI) and Maximum-Likelihood (ML) method. The best-fit model of sequence evolution was identified in MEGA v.799, and the GTR + G + I model was selected. The BI analysis was performed in MrBayes v.3.2.6100,101, and the ML analysis was conducted in PhyML v.3.0102. Parameter settings were previously described by Androsiuk et al.26.



6 




Pi   [Figure 5](https://www.nature.com/articles/s41598-020-68563-5/figures/5)




- #### [Molecular phylogenetics provides new insights into the systematics of Pimelea and Thecanthes (Thymelaeaceae)](https://www.publish.csiro.au/SB/SB16013) 



`Partitioning the dataset is an important step in phylogenetic
analyses, because this process is known to affect the accuracy of
phylogenetic inference (Lanfear et al. 2012). Accordingly, we
selected the best-fitting partitioning scheme by using the
greedy search algorithm in PartitionFinder ver. 1.1.1 (Lanfear
et al. 2012). The optimal partitioning scheme split the sequence
alignment into six subsets (Table 2).`





- #### [Question: extract annotation for mVISTA whole genome comparison](https://www.biostars.org/p/364641/) | [Answer](https://blogs.fu-berlin.de/gruenstaeudl/2020/10/12/buena-vista-con-mvista/)


Buena vista con mVISTA
For an upcoming publication, a doctoral student and I want to visualize the sequence variability among several plastid genomes via the tool mVISTA. This tool is often employed in plastid phylogenomic studies and generally simple to use. However, if a user wishes to input custom annotations, it can be quite tricky to generate the correct input for the software. Hence, I automated the generation of the input files as described below.

An effective visualization of plastid genomes in mVISTA requires the x input genomes the user wants to visualize („input genomes“ hereafter; e.g., x=2) as well as a reference genome. For simplicity, a user could employ x1 as the reference genome. The input genomes should be in FASTA format, the reference genome in GenBank format.

To generate custom annotations based on the reference genome, I employ the following Bash code, which (a) converts the reference genome from GenBank format to a cleaned GFF3 format (and incidentally also saves the genome in FASTA format), and (b) converts the cleaned GFF3 file to the mVISTA input.

```
NF=NC_000932.gb # Just an example
## Converting input file from GenBank format to a cleaned GFF3 format
# Note: This step also generates a FASTA file
grep -vE "codon_start|db_xref|exception" $INF > ${INF}2
to-gff --getfasta ${INF}2 ${INF%.gb*}.gff
grep "gene=" ${INF%.gb*}.gff | \
    awk -F';' '{print $1}' | \
    grep -vE "remark|intron|misc_feature|repeat_region" > ${INF%.gb*}.gff.clean
rm ${INF}2 ${INF%.gb*}.gff
## Converting input file from GFF3 format to VISTA format
grep -v "^#" ${INF%.gb*}.gff.clean | \
    grep -v "rps12" | \
    awk '{if ($3 ~ /gene/) {print $7" "$4" "$5" "$3" "$9} else {print $7" "$4" "$5" "$3} }' | \
    awk '{if ($4 ~ /gene/) {gsub(/\+/, ">", $1); gsub(/\-/, "<", $1); print $0} else {$1=""; print $0} }' | \
    sed 's/gene=//' | \
    sed 's/gene/agene/' | \
    sort -n -k2 | \
    sed 's/agene/gene/' | \
    awk '{$1=$1}1' | \
    sed 's/CDS/exon/' | \
    sed 's/tRNA/utr/' | \
    sed 's/rRNA/utr/' > ${INF%.gb*}.mvista


```







- #### My ultimate analyses reference from Bioinformaticians of Indian | [Trichopus zeylanicus](https://acsess.onlinelibrary.wiley.com/doi/full/10.3835/plantgenome2019.04.0032)



 - #### [The complete chloroplast genome of Gleditsia sinensis and Gleditsia japonica: genome organization, comparative analysis, and development of taxon specific DNA mini-barcodes](https://www.nature.com/articles/s41598-020-73392-7#Sec14)

- #### [The Complete Chloroplast Genome Sequences of 14 Curcuma Species: Insights Into Genome Evolution and Phylogenetic Relationships Within Zingiberales](https://www.frontiersin.org/articles/10.3389/fgene.2020.00802/full)


- #### [GOOD PLANT PLASTOME ANNOTATION TOOL](https://chloe.plantenergy.edu.au/annotate.html)  | THESIS [HERE](https://research-repository.uwa.edu.au/en/publications/assembly-annotation-and-analysis-of-chloroplast-genomes)

- #### [un bon argumentaire ..le graph des chloroplast la avec la perte du gene en question](https://peerj.com/articles/9315/)

- #### [Le graph de dn/ds et le choix de l'assemblage](https://www.mdpi.com/2223-7747/9/11/1497)

- #### [Simple et clair a suivre ](https://www.frontiersin.org/articles/10.3389/fpls.2018.01989/full)

- ##### vraiment interessant paper chloroplast SNP origins geography [paper](https://www.frontiersin.org/articles/10.3389/fpls.2019.00334/full#h3)

- #### [commentaire des resultats](https://www.mdpi.com/2223-7747/9/10/1332)

- #### Use of novoplasty for chloroplast assembly | [Example 1](https://www.frontiersin.org/articles/10.3389/fgene.2019.00444/full#h3) | [Example 2](https://www.frontiersin.org/articles/10.3389/fpls.2020.00729/full#h3) | [Example 3](https://www.hindawi.com/journals/bmri/2020/8542797/) | [Example 4](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6253-5) | [Example 5](https://www.nature.com/articles/s41598-018-35654-3#Sec2) | [Example 6](https://www.nature.com/articles/s41598-019-48708-x#Sec10) | [Example 7](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6646306/) | [Example 8](https://www.mdpi.com/1422-0067/19/8/2443/htm) | [Example 9](https://link.springer.com/article/10.1007/s11033-020-05567-4?utm_source=other) | [Example 10](https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/tpg2.20019)

- #### Distribution de mes samples

- ##### [Sesamum indicum](http://www.plantsoftheworldonline.org/taxon/urn:lsid:ipni.org:names:675971-1)

- ##### [Sesamum alatum](http://www.plantsoftheworldonline.org/taxon/urn:lsid:ipni.org:names:675950-1)

- ##### [Sesamum angolense](http://www.plantsoftheworldonline.org/taxon/urn:lsid:ipni.org:names:675951-1)

- ##### [Sesamum radiatum](http://www.plantsoftheworldonline.org/taxon/urn:lsid:ipni.org:names:675994-1)

- ##### [Sesamum pedaloides](https://books.google.co.kr/books?id=lUOWDwAAQBAJ&pg=PA301&lpg=PA301&dq=Sesamum+pedaloides&source=bl&ots=65fqz989JJ&sig=ACfU3U1Z7J8P9tl3NvSS1wtBmHbdpb4rRg&hl=fr&sa=X&ved=2ahUKEwjnrbr26KHsAhWrIqYKHW5gB9cQ6AEwDnoECAUQAg#v=onepage&q=Sesamum%20pedaloides&f=false) | Page 25



- ##### Parfait exemple using chloroplast to solve taxonomic dispute | Idee Map pour ton article | [Taxonomy and biogeography of Diapensia (Diapensiaceae) based on chloroplast genome data](https://onlinelibrary.wiley.com/doi/full/10.1111/jse.12597?casa_token=RqchIwzluSQAAAAA%3AyULyoTr8U-Pm0JS22W44Igy3H1Hs1tV-xyoy0T8SM-ktTXsBBz9dMz0sL2syNYUbSLrMN1iUgnmSV062)


- #### 2020 Bel argument en faveur du choix des CDs pour le phylogenetic analysis 

> Only the homologous CDs were used to construct the phylogenetic relationship to reduce data redundancy...

[Chloroplast genome features of an important medicinal and edible plant: Houttuynia cordata (Saururaceae)](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0239823)


- #### 2020 Wahooo nice surprise | Only One species in Genomis Elselvier | [Characterization of Withania somnifera chloroplast genome and its comparison with other selected species of Solanaceae](https://www.sciencedirect.com/science/article/pii/S0888754319303258)

- #### A second example from Genomics elselvier | [Chloroplast genome sequences of Artemisia maritima and Artemisia absinthium: Comparative analyses, mutational hotspots in genus Artemisia and phylogeny in family Asteraceae](https://www.sciencedirect.com/science/article/pii/S0888754319304537#!)

- #### [A third example: Evolutionary dynamics of chloroplast genomes in subfamily Aroideae (Araceae)](https://reader.elsevier.com/reader/sd/pii/S0888754319308420?token=02DF5B01DEF65C2E1F4696ADFC4E0C5757E302859EB016BCA306C9882B6983078C077A62658FF6B41F90EB878CDE0F2D)


- #### Very nice argument | `` Previous studies based on plastid fragments and/or nuclear ribosomal DNA have had limited success resolving relationships within the genus Salvia. This study evaluates the efficacy of complete plastome sequences for phylogenetic inference within Salvia, using the recently established Salvia subg. Glutinaria as a case study. We use these plastomes to identify hypervariable and simple sequence repeat (SSR) regions for future studies within Salvia.`` [The Chloroplast Genome of Salvia: Genomic Characterization and Phylogenetic Analysis](https://www.journals.uchicago.edu/doi/pdf/10.1086/710083)


- #### 2020 Only two | Very nice R graph [Comparative analysis of chloroplast genomes in Vasconcellea pubescens A.DC. and Carica papaya L.](https://www.nature.com/articles/s41598-020-67943-1)
- #### 2020 Like my case [Evolutionary and phylogenetic aspects of the chloroplast genome of Chaenomeles species](https://www.nature.com/articles/s41598-020-67943-1)

- #### [The Complete Chloroplast Genome Sequence of the Speirantha gardenii: Comparative and Adaptive Evolutionary Analysis](https://www.mdpi.com/2073-4395/10/9/1405)

- #### [Species-tree inference with BEAST 2 and SNAPP](http://evomics.org/learning/population-and-speciation-genomics/2020-population-and-speciation-genomics/species-tree-inference/)
- #### [IRscope creator plastome paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0196069)

- #### [inverted repeat structural evolution in Rhus L](https://peerj.com/articles/9315/)

- #### [Simple tuto](https://www.nature.com/articles/s41598-020-59204-y.pdf)

- #### [Interessant title](https://www.nature.com/articles/s41598-020-68563-5#Sec11)


- #### [Insights Into Chloroplast Genome Evolution Across Opuntioideae (Cactaceae) Reveals Robust Yet Sometimes Conflicting Phylogenetic Topologies](https://www.frontiersin.org/articles/10.3389/fpls.2020.00729/full#B91)

- #### [Best Practices/Softwares To Calculate Ka/Ks Ratio](https://www.biostars.org/p/5817/)

- #### [The complete chloroplast genome of Stryphnodendron adstringens (Leguminosae - Caesalpinioideae): comparative analysis with related Mimosoid species](https://www.nature.com/articles/s41598-019-50620-3)

- #### [The Complete Chloroplast Genome Sequences of the Medicinal Plant Forsythia suspensa (Oleaceae)](https://www.mdpi.com/1422-0067/18/11/2288)

- #### [Comparative Analysis of the Complete Chloroplast Genomes of Five Quercus Species](https://www.frontiersin.org/articles/10.3389/fpls.2016.00959/full)

- #### [IR CONTRACTION](https://www.nature.com/articles/s41438-019-0171-1#:~:text=Compared%20with%20the%20nuclear%20genome,for%20diversity%20and%20evolution%20studies.)

- #### [PSEUDOGENE](https://peerj.com/articles/2699/?utm_source=TrendMD&utm_campaign=PeerJ_TrendMD_1&utm_medium=TrendMD)

- #### [Inspiration](https://www.researchsquare.com/article/rs-20050/v1)

- #### [Dr swati 1](https://peerj.com/articles/9448/)

- #### [Dr swati 2](https://www.mdpi.com/2223-7747/9/5/568/htm)



# R spatial

- #### [r map tuto 1](https://datavizm20.classes.andrewheiss.com/example/12-example/) [content](https://datavizm20.classes.andrewheiss.com/content/) [video](https://www.youtube.com/watch?v=Opzwtegvuv4&t=743s)
- #### [r map tuto 2 create a shp file from data frame ](https://datacarpentry.org/r-raster-vector-geospatial/10-vector-csv-to-shapefile-in-r/)
- #### [r map tuto 3 raster manip](https://datacarpentry.org/r-raster-vector-geospatial/)
- #### [r map tuto 4](https://www.earthdatascience.org/courses/earth-analytics/spatial-data-r/make-maps-with-ggplot-in-R/)
- #### [r map tuto 5](http://zevross.com/blog/2018/10/02/creating-beautiful-demographic-maps-in-r-with-the-tidycensus-and-tmap-packages/)
- #### [r map tuto 6 sf test](https://gist.github.com/andrewheiss/8e514355d75aac1a7d6b91a67b896f20) [fig](https://twitter.com/andrewheiss/status/1291431466556100608) [data](https://github.com/jvangeld/ME-GIS)
- #### [reading and writing spatial data in r in r](https://mgimond.github.io/Spatial/reading-and-writing-spatial-data-in-r.html)
- #### [Drawing beautiful maps programmatically with R, sf and ggplot2 — Part 2: Layers](https://www.r-spatial.org//r/2018/10/25/ggplot2-sf-2.html)
- #### [Geospatial Health Data: Modeling and Visualization with R-INLA and Shiny](https://www.paulamoraga.com/book-geospatial/)


# Rogeria longiflora
 
- #### March 2012 | [Germination and growth requirements of Rogeria longiflora – Medicinal plant of the Namib Desert](https://www.sciencedirect.com/science/article/pii/S0254629911001797)
- #### [RARE AND INTERESTING PLANTS OF THE NAMIB DESERT](https://journals.co.za/docserver/fulltext/veld/80/4/2395.pdf?expires=1599199452&id=id&accname=guest&checksum=A132124BA5604D1262C81C2E69E6C821)
- #### [Phylogeny of Pedaliaceae and Martyniaceae and the Placement of Trapella in Plantaginaceae s. l.](https://www.ingentaconnect.com/content/aspt/sb/2015/00000040/00000001/art00030)
- #### [Pedaliaceae](https://link.springer.com/chapter/10.1007/978-3-642-18617-2_17)
- #### [Wood Anatomy of Martniaceae and Pedaliaceae](https://scholarship.claremont.edu/cgi/viewcontent.cgi?referer=https://scholar.google.com/&httpsredir=1&article=1506&context=aliso)
- #### [Dispersal Biology of Desert Plants : Restriction of Dispersal in Time](https://link.springer.com/chapter/10.1007/978-3-662-03561-0_8)
- #### [Dewinteria](http://opus.sanbi.org/bitstream/20.500.12143/4207/2/van%20Jaarsveld_2007_PEDALIACEAE%2C%20Dewinteria%2C%20a%20new%20semisucculent%2C%20cliff-dwelling%20genus%20endemic%20to%20the%20Kaokoveld%2C%20Namibia.pdf)
- #### [Ecogeography and taxonomy of Rogeria J. Gay ex Delile (Pedaliaceae)](https://www.tandfonline.com/doi/abs/10.1080/00837792.2013.867609)
- #### [Rogeria species](http://www.ville-ge.ch/musinfo/bd/cjb/africa/resultat.php?efNom=rogeria&efFamille=&projet%5B%5D=FSA&projet%5B%5D=FTA&projet%5B%5D=FNA&projet%5B%5D=BDM&langue=an&pbRecherche=Search)


# Ceratotheca triloba

- #### [basic description](https://onlinelibrary.wiley.com/doi/pdf/10.1111/j.1467-8748.2011.01747.x?casa_token=GyWXoq2XXfMAAAAA:IPI77qhB7znxhHSl3EASdnGYbxNrctsA7U-YE84wYtiYgLhOBCqyT_mXABhp056dOyQLZTymddbgUunT)

# Sesamum alatum

- #### [Inspirationnal paper 1](https://www.sciencedirect.com/science/article/pii/S0888754318306591)


# Cucumis 


- #### [Syntenic relationships between cucumber (Cucumis sativus L.) and melon (C. melo L.) chromosomes as revealed by comparative genetic mapping](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-12-396)
- #### [Cucumber (Cucumis sativus) and melon (C. melo) have numerous wild relatives in Asia and Australia, and the sister species of melon is from Australia](https://www.pnas.org/content/107/32/14269)
- #### [Pumpkins, cucumbers, and watermelons diverged from a single melon ancestor](https://geneticliteracyproject.org/2017/10/13/pumpkins-cucumbers-watermelons-diverged-single-melon-ancestor-study-shows/)
- #### [RELATIONSHIPS OF CUCUMBERS AND MELONS UNRAVELED:MOLECULAR PHYLOGENETICS OF CUCUMIS AND RELATED GENERA (BENINCASEAE, CUCURBITACEAE)](https://bsapubs.onlinelibrary.wiley.com/doi/pdfdirect/10.3732/ajb.94.7.1256)
- #### [Comparative Analysis between Wild and Cultivated Cucumbers Reveals Transcriptional Changes during Domestication Process](https://www.mdpi.com/2223-7747/9/1/63/htm)
- #### [QTL mapping and genome-wide association study reveal two novel loci associated with green flesh color in cucumber](https://link.springer.com/article/10.1186/s12870-019-1835-6)


# Fungal Genomics 


- #### [Great inspiration discovered today 16 december 2020 on research square |  Genome sequencing and comparative genomic analysis of highly and weakly aggressive strains of Sclerotium rolfsii, the causal agent of peanut stem rot](https://assets.researchsquare.com/files/rs-38224/v2/9acd3546-03af-4b93-8e48-085d43b35327.pdf)

- #### [FocSge1 in Fusarium oxysporum f. sp. cubense race 1 is essential for full virulence](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-020-01936-y)

- #### [ONT assembly with Canu with mitogenome resolution](https://academic.oup.com/gigascience/article/9/9/giaa099/5908739)
- #### [Inspi](https://link.springer.com/article/10.1186/gb-2010-11-7-r73)

- #### [Inspiration paper 1 for Graph color respect with R  ](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-06871-w)
- #### [Inspirational paper 2](https://imafungus.biomedcentral.com/articles/10.1186/s43008-019-0011-9)
- #### [Inspirationnal paper 3](https://www.nature.com/articles/s41598-018-30335-7)
- #### [Code and script Fungal annotation assemblies and so on](https://gitlab.gwdg.de/alice.feurtey/genome_architecture_zymoseptoria)
- #### [Mechanism of pathogenicity](https://books.google.fr/books?hl=fr&lr=&id=wGgCEAAAQBAJ&oi=fnd&pg=PA185&dq=NBS-LRR+gene+cloning+plant+disease+resistance&ots=pwwZ21T0-W&sig=yZRXOuy_gzvH6aC-TSjfgZoU0Sg#v=onepage&q=NBS-LRR%20gene%20cloning%20plant%20disease%20resistance&f=false)

- #### [Grouped barchart customization](https://www.r-graph-gallery.com/48-grouped-barplot-with-ggplot2.html)
- #### [Change label order](https://www.datanovia.com/en/fr/blog/ggplot-comment-changer-lordre-des-legendes/)


# NNS-LRR genes 

- #### [Inspirational paper 1](https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-020-02576-0#Sec16)
- #### [Dioscorea](https://www.frontiersin.org/articles/10.3389/fgene.2020.00484/full)
- #### [Cassava](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4422547/pdf/12864_2015_Article_1554.pdf)
- #### [Inspirational paper 2](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0239275)



# Linkage mapping

- #### [High-density linkage map reveals QTL for Type-I seed coat cracking in RIL population of soybean [Glycine max (L.) Merr.]](https://link.springer.com/article/10.1007/s10681-020-02684-w)

# Genetic diversity

- #### [Diversity analysis of 80,000 wheat accessions reveals consequences and opportunities of selection footprints](https://www.nature.com/articles/s41467-020-18404-w) | [Code github](https://github.com/jfranco1951/Genetic-Diversity)


# Dotplot RNA

- #### [ Dotplot RNA](https://www.biostars.org/p/327039/)




# Doudou Sylla

#### [graph 1](https://www.google.com/search?q=GGPLOT2+FACET&rlz=1C1EJFC_enKR900KR900&sxsrf=ALeKk03R6Jk9D47FDaWh9PPny2tn9Nu_MQ:1600266205845&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj99PqV8O3rAhWRMd4KHVIqAkQQ_AUoAXoECA4QAw&biw=1920&bih=920#imgrc=5k-afjZeAPxLkM&imgdii=kWtsYbSNy82DSM) | [barplot](http://www.sthda.com/french/wiki/ggplot2-barplots-guide-de-demarrage-rapide-logiciel-r-et-visualisation-de-donnees) | [cord_flip](http://www.sthda.com/french/wiki/ggplot2-barplots-guide-de-demarrage-rapide-logiciel-r-et-visualisation-de-donnees) | [faceting](https://www.r-graph-gallery.com/223-faceting-with-ggplot2.html)  | [example](https://stackoverflow.com/questions/34001024/ggplot-order-bars-in-faceted-bar-chart-per-facet)



#### Tuto colinearity [MCScanX](https://github.com/wyp1125/MCScanX)


##### Step 1:  Create a m8 format BLASTP file

I assume I have Arabidopsis and xyz plant genome for colinearity work

So, I should make a database for 

First, we need to tell BLAST that the Arabidopsis sequences are (a) a database, and (b) a protein database. That’s done by calling ‘makeblastdb’:

```
makeblastdb -in Ara.protein.faa -dbtype prot

```

then run blastall


```

$blastall  -i  myxyz_query_file_protein.fasta  -d Ara.protein_database -p blastp -e 1e-10 -b 5 -v 5 -m 8 -o xyz.blast


```


For MSCsanX  you should have an output lin m8 [format](https://edwards.sdsu.edu/research/blast-output-8/). An example [here](https://raw.githubusercontent.com/wyp1125/MCScanX/master/data/at.blast)


```

AT1G50920   AT1G50920    100.00  671     0       0       1       671     1       671     0.0     1316

```


Make sure you have this format before downstream analysis



## Step 2: Create a simplified "gff" file as inputs

I need to install bedops in order to parse the gff files. [A biostar resource for this operation](https://www.biostars.org/p/61386/)

I will use conda since I do not have a rot access.

I will set a specific environment.


```

conda create -n bedops_env


```

Then activate the environment

```
source activate bedops_env

```

Then install the tool

```

conda install -c bioconda bedops


```

Now use the tool to parse my gff file. A nice help is [here](https://bedops.readthedocs.io/en/latest/content/reference/file-management/sorting/sort-bed.html)


```

sortBed -i myfile.gff | gff2bed > my_sorted_file.bed

```

The xyz.bed file holds gene positions, following a tab-delimited format like [this](https://github.com/wyp1125/MCScanX/blob/master/data/at.gff):


```
chr#    starting_position       ending_position gene

```

Note: for chr#, a two-letter short name is used as prefix for the species; # is the chromosome number. (For example, the second chromosome of Arabidopsis thaliana should be denoted as at2.) The bed format is defined here, and is especially useful since there are a ton of tools that can handle bed files, most notably BEDTOOLS. The xyz.bed file can be generated by parsing the .gff3 file released by the sequencing initiatives. Repeat of the same gene is not allowed in the .bed file. When comparing multiple genomes, simply concatenate all inter-/intra-species m8 blast output into xyz .blast file and concatenate all gene positions of different species into xyz.bed file.

It is advised that to make MCscanX generate more reasonable results, the number of BLASTP hits for a gene should be restricted to around top 5. When you have xyz.blast and xyz.bed ready, put them in the same folder. Then you can simply use:

```

$ ./MCScanx  dir/xyz

```

[here](https://youtu.be/KMlj8CGnB2c?t=477) a video on how to run it.



possible to calculate ka ks with bioperl with the .syntheny output of MCscan. follo this [tuto](possible to calculate ka ks with bioperl with the .syntheny output of MCscan 
)

#### Note

A guy I know use a four step logic

- Step1: get coliniarity gene wih MCScanX(https://github.com/wyp1125/MCScanX)

- step2: Multiple sequence alignement of those colinear genes with ClustalW2

- step3: Convert into codon WITH [pal2nal](http://www.bork.embl.de/pal2nal/) | [pal2nal](https://figshare.com/articles/An_example_of_PAL2NAL_input_and_output_files/9192)

- step4: calculate ka ks with CODEML of [PAML](https://www.protocols.io/view/introduction-to-calculating-dn-ds-ratios-with-code-qhwdt7e) | [EasyCodeML: A visual tool for analysis of selection using CodeML](https://onlinelibrary.wiley.com/doi/full/10.1002/ece3.5015)  | [github](https://github.com/BioEasy/EasyCodeML)  | [download](https://github.com/BioEasy/EasyCodeML/releases/tag/1.0)

- step5: plot the density graph in R WITH GGPLOT2

#### TIME DIVERGENCE



Here a [resource](http://www.timetree.org/resources).


My guy use Reltime estination methode with MEGA  after Single copy alignment with MAFFT

He download some calibration files on timetree.org and run I think

a good tuto for RELTIME time divergence estimation with [MEGAx](https://www.megasoftware.net/pdfs/mello_2018.pdf)

for the calibration he retieve from timetrre.org database 



[a nice and complete tutorial video](https://www.youtube.com/watch?v=x5TqC5okZWo)

[Inferring selection with MEGA](https://www.youtube.com/watch?v=wNLZoKkdJ38)


un nice tuto sur cette affaire de time tree et autre [ici](https://kor.kyhistotechs.com/complex-jujube-genome-provides-insights-into-fruit-tree-biology-31050929)


#### Visualize syntheny


Now I want to visualize the syntheny. 

[Synvisio is awesome](https://github.com/kiranbandi/synvisio)

I just need two files:

- The simplified gff file that was used as an input for a McScanX query.
- The collinearity file generated as an output by McScanX for the same input query.


Possible to do that [online](https://synvisio.github.io/#/)



in case of troubles for making bed file please Yedomon go to this [page](https://github.com/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29#workflow)

Instal jcvi [conda package manager](https://anaconda.org/bioconda/jcvi) 

and follow [this plan](https://github.com/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29#workflow)

jvci is a python version of MCScan ..so you can use it also. And jvci can help you to render nice visualization  




I just discover a very nice and helpfull website for some [sofwares](https://sites.google.com/view/dr-wasim/softwares) , [protocol](https://sites.google.com/view/dr-wasim/protocols/nucleic-acid-extraction) , [tutorials](https://sites.google.com/view/dr-wasim/bioinformatic-tools/tbtools)

# Whole genome duplication



[nice paper](https://www.nature.com/articles/s41467-020-18771-4)

Can be proved through ka?ks analysis graph, syntheny graph, phylogenetic tree time divergence analysis... Example with this [paper](https://onlinelibrary.wiley.com/doi/full/10.1111/1755-0998.13261#men13261-fig-0003) and this [paper](https://onlinelibrary.wiley.com/doi/full/10.1111/pbi.13350)

[Good review](https://www.cell.com/trends/plant-science/fulltext/S1360-1385(18)30159-6)




[How to parse orthofinder results?](https://www.biostars.org/p/366002/)



#### How to download genome data from ncbi

I wanted to download bacteria genome assembly from ncbi, but I did not know how to do.

By googling I fortunately found this [link](https://www.ncbi.nlm.nih.gov/datasets/docs/command-line-assembly/)

They recommand to install the ncbi program named datasets. How to install it ? The response is [here](https://www.ncbi.nlm.nih.gov/datasets/docs/rehydrate/).

It is possible to downmload a hude dataset from a taxon with NCBI taxon ID. 

After installation type ./datasets download genome taxon --help to see the details arguments

an other way to do is to [here](https://astrobiomike.github.io/unix/ncbi_eutils) and [here](https://dmnfarrell.github.io/bioinformatics/assemblies-genbank-python)



#### SP work

- #### [Inspiration 1](https://peerj.com/articles/9448/)

- #### [Inspiration 2](https://link.springer.com/article/10.1007/s13205-020-02372-5)

#### Population genomics work 

##### Sesame

- #### [Cui et al 2017](https://www.frontiersin.org/articles/10.3389/fpls.2017.01189/full)
- #### [Mei et al 2017](https://www.frontiersin.org/articles/10.3389/fpls.2017.00636/full)

##### Others
- #### [Perfect example](https://www.cell.com/molecular-plant/fulltext/S1674-2052(18)30343-5)
- #### [Jujube example](https://onlinelibrary.wiley.com/doi/10.1111/pbi.13480)
- #### [Wheat](https://link.springer.com/article/10.1186/s12864-020-06835-0#Sec14)
- #### [Joukhadar et al 2017](https://www.frontiersin.org/articles/10.3389/fpls.2017.02115/full#h3)
- #### [Wang et al 2018](https://www.cell.com/molecular-plant/fulltext/S1674-2052(18)30187-4#secsectitle0020) | Domestication area
- #### [Good journal for publication](https://gsejournal.biomedcentral.com/articles/10.1186/s12711-020-00581-3#Sec2)

# Homework seed science

The idea is to use wild genome for rice sedd nutritionnal quality improvmen through genomics breeding

Some references

- #### [Potentiality of Wild Rice in Quality Improvement of Cultivated Rice Varieties](https://link.springer.com/chapter/10.1007/978-981-15-4120-9_4)

- #### [Germplasm and Genetic Diversity Studies in Rice for Stress Response and Quality Traits](https://link.springer.com/chapter/10.1007/978-981-15-4120-9_3)
- #### [Rice Grain Quality and Abiotic Stress: Genomics and Biotechnological Perspectives](https://link.springer.com/chapter/10.1007/978-981-15-4120-9_30)
- #### [argument important of reseq for agronomic traits](https://link.springer.com/chapter/10.1007/978-981-15-4120-9_6)

- #### [genome wild rufipogon](https://www.nature.com/articles/s42003-020-0890-8#Sec8) | [another version](https://link.springer.com/article/10.1007/s11427-020-1738-x)
- #### [proteomics analysis](https://link.springer.com/article/10.1186/s12953-014-0051-4)
- #### [Improvement of Nutritional Quality of Rice Seed Through Classical Breeding and Advance Genetic Engineering](https://link.springer.com/chapter/10.1007/978-981-15-5337-0_23)
- #### [nICE GRAPH R ](https://nph.onlinelibrary.wiley.com/doi/full/10.1002/ppp3.10151)
# Karyotyping paper

- #### [High impact factor potential journal: Journal of Genetics and Genomics](https://www.journals.elsevier.com/journal-of-genetics-and-genomics)
- #### [Example IF = 6.141 ](https://onlinelibrary.wiley.com/doi/full/10.1111/tpj.14536)
- #### [Example IF = 2.076](https://academic.oup.com/botlinnean)
- #### [Example 2020 Frontiers in Plant Science IF = 4.402](https://www.frontiersin.org/articles/10.3389/fpls.2020.00569/full)
- #### [Example 0: IF = 5.065 | Construction and application of oligo-based FISH karyotype of Haynaldia villosa](https://www.sciencedirect.com/science/article/pii/S1673852718301164)
- #### [Example 1: IF = 3.395 | The Crop Journal ](https://www.sciencedirect.com/science/article/pii/S2214514116300484)
- #### [Example 2: IF = 1.085 | Biochemical Systematics and Ecology](https://www.sciencedirect.com/science/article/pii/S0305197815002203)
- #### [Example 2: IF = 1.662 | Frontiers in Life Science](https://www.tandfonline.com/doi/full/10.1080/21553769.2015.1041166)
- #### [Example 3: IF = NA | Plant Breed. Biotech.](http://www.plantbreedbio.org/journal/view.html?volume=7&number=3&spage=237&year=2019)

# Download nt and nr from ncbi

[solution](https://www.biostars.org/p/387902/)


# Review : Understanding African plant diversification: insights from special plant species complexes in sub-Saharan Africa


- #### [Cradles and museums of generic plant diversity across tropical Africa](https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.16293#nph16293-bib-0016)

- #### [Beyond trees: Biogeographical regionalization of tropicalAfrica](https://onlinelibrary.wiley.com/doi/pdf/10.1111/jbi.13190?casa_token=4NHZao-oL4cAAAAA:AXDZLPyk6aNkZl2PnqV1Fq7tKXYot7wYp5LS0hZOQvRWauK5_zf1ceSOMEUpSJOEmQH4X_3YK7e2YTj8)


- #### [PLANT SYSTEMATICS IN SOUTH AFRICA: A BRIEF HISTORICAL OVERVIEW, 1753–1953](https://www.tandfonline.com/doi/abs/10.1080/00359199909520411?journalCode=ttrs20)

- #### [Floristics of the angiosperm flora of Sub-Saharan Africa: an analysis of the African Plant Checklist and Database](https://www.researchgate.net/publication/233681713_Floristics_of_the_angiosperm_flora_of_Sub-Saharan_Africa_an_analysis_of_the_African_Plant_Checklist_and_Database)


# Earlier African systematics investigation

The African continent is composed of more than 45,000 species distributed on a area of 29 million km2 (Klopper, R. R., Gautier, L., Chatelain, C., Smith, G. F., and Spichiger, R. (2007). Floristics of the angiosperm flora of sub-Saharan Africa: an analysis of the African plant checklist and database. Taxon 56, 201–208.). Early 1600's, diverse European botanists started the documentation of African flora. Individual investigations based on diverses schools dependending on the practitionners were used. However, the traceability of the collected data and the lack of communication make difficulta comprehensive knowledge of the African plants classification. Plant systematics in Africa has evolved a dramatic turn with the Association pour l'Etude Taxonomique de la Flore d'Afrique Tropicale (AETFAT) (Association for the taxonomic study of the flora of tropical Africa in English).





# How I solved ALI TV installation in my Centos 7


```
sudo yum install perl-devel

sudo yum install cpanminus

sudo cpanm Bio::Perl # to install bioperl tools

sudo cpanm --force Bio::Perl # force failed installation module

```



I got the solution [here](https://www.biostars.org/p/345331/)

to visualize  ...... https://alitvteam.github.io/AliTV/d3/AliTV.html


awesome tuto [here](https://alitvteam.github.io/AliTV/gcb_2016)


# Nucmer alignement artemis

Hi Yedomon,

 

Below is the command for nucmer, delta-filter, show-coords, comparison file. I hope you can make it. 

 

the lines not started with # are the command you will use. 

#Let's suppose you have a delta file called "gb_zz.delta", which was produced by nucmer with ref=zz, query=gb.

#Do show-coords with -cTrl option like below;

show-coords -cTrl gb_zz.delta > gb_zz.delta.coords

#or do delta-filter first with %match (-i option) >=95% and match length (-l option) >=500bp

delta-filter -i 95 -l 500 gb_zz.delta > gb_zz_i95l0.5k.delta

#then do show-coords like above.
#Now let's make a comparison file for ACT like below;

more gb_zz.delta.coords | awk '$0~/^[0-9]/' | awk '{print $5,$7,$1,$2,$12,$3,$4,$13}' > gb_zz.cmp

=====below is not a commend
#where each column has;
$5=alignment_len of reference
$7=%id
$1,$2,$12=ref start, end, Ref_name (always +)
$3,$4,$13=qry start, end, Qry_name (can be +/-)

#load gb_zz.cmp file to ACT
sequence1=reference
sequence2=query
comparison_file=gb_zz.cmp


 

Let me know if you have any problem. Good luck.

 

Yeisoo


One day I face a problem ....when running a well install programm under linux.


The problem was slove using the command: 


chown -R user:user folder/


Permission issue was the problem

# Trover un moyen de gerer les gaps surtout leur positions dans une sequence

-Une reponse [ici](https://stackoverflow.com/questions/49303791/finding-the-positions-and-lengths-of-gaps-indels-in-a-sequence-alignment-with) que je peux adapter a mon cas en remplacant le - par N

 comme par hasard je decouvre ceci..comment visualiser la proportion de gap dans un genome. tres interessant. site [ici](https://bioinformaticsworkbook.org/dataWrangling/R/visualize-gaps-in-genomes.html#gsc.tab=0)
 
 
 
 ### Pour record ram max installer yime em faisant yum install time puis lancer votre code comme ceci /usr/bin/time -o out.time.ram.txt -v votrecommande &> log & 
 
 
 # Back-to-back bar graph 
 
 - #### [video](https://www.youtube.com/watch?v=DbFU7Js8_h4)
 
 - #### [Inspi color](https://www.alanapirrone.com.au/blog/tag/back-to-back+bar+chart)


# ---Comment jai pu avoir la list des data pacbio nanopore sur ENA. JUST TAPER FUSARIUM OXYSPPORUM PACBIO PUIS TELECHARGER LE FICHIER XML PUIS CONVERTIR CE FICHIER SUR CE [SITE](https://conversiontools.io/convert/xml-to-excel)



##### Translate english file into french [here](https://www.onlinedoctranslator.com/app/translationprocess)


##### How I solved this error message when installing Hifiasm `make: g++: Command not found`  I found the solution [here](https://gahee0416.tistory.com/23) by doing `sudo yum install gcc-c++` since I am using Centos OS 7 


###### Phylogenetic tree visualization 


- #### [nice tuto](http://evomics.org/wp-content/uploads/2016/06/TreeEditingVisualization_WoPhylogenomics_CK2017.pdf)



##### A work for Yolande

```


# Libraries
library(ggplot2)


#theme_set(theme_minimal())

#--Data importation
data_humidity = read.csv("data_humidity.csv", sep = ";", h=T , dec = ",")

data_new <- data_humidity                               # Replicate data
data_new$Profondeur <- factor(data_new$Profondeur,      # Reordering group factor levels
                         levels = c("10cm", "20cm", 
                                    "30cm", "40cm", 
                                    "50cm", "60cm",
                                    "70cm", "80cm", 
                                    "90cm", "100cm", 
                                    "110cm", "120cm",
                                    "130cm", "140cm",
                                    "150cm", "160cm"))

#--Plot

p <- ggplot(data_new, aes(x=as.Date(Date), y=Humidite)) +
  geom_line(color="steelblue") + 
  geom_point(color="steelblue", alpha = 1/10) +
  xlab("") +
  theme_bw()+
  #theme_minimal() +
  theme(axis.text.x=element_text(angle=60, hjust=1)) +
  scale_x_date(limit=c(as.Date("2018-07-31"),as.Date("2019-11-07"))) +
  ylim(0,30)+
  facet_wrap(~Profondeur)+
  labs(y = "Soil moisture (mm)", x = "Date")

p






```

[output1](https://github.com/Yedomon/Bric_a_Brac/blob/master/Plot1_6.91_6.91.pdf)

[output2](https://github.com/Yedomon/Bric_a_Brac/blob/master/Plot2_6.91_6.91.pdf)

[dataset](https://github.com/Yedomon/Bric_a_Brac/blob/master/data_humidity.csv)



##### A serious issue today. Just want to get cds translated protein fron annonated genbank file but I struggle alot


Here is the solution.

I gound it [here](https://pypi.org/project/gbseqextractor/).

`gbseqextractor` can do many things...


So I install the tool by doing 


`
pip install gbseqextractor

`


Then I run

```

 gbseqextractor -f kc.gb -prefix china_cds -seqPrefix yuzhi -types CDS -cds_translation


```

I got the cds fatsa nucleotide as well as protein file. Awesome.



##### Calculate Pi for my chloroplast datasets.

I have five assemblies.

I used MAFFT to aligne nucleotide assembly files

then import into DNAsp v6


then GO TO ANALYSIS > DNA POLYMORPHISM

SELECT Pi and click on compute button in order to change The step size  to 50 base pairs, and window length  to 800 base pairs.


The default one give a very narrow bin


After getting the result use ggplot2 to draw the graph of pi following the sliding window following this [tutorial](https://www.r-graph-gallery.com/line_chart_annotation.html)



I found this basic plot [also](https://www.r-graph-gallery.com/200-change-color-in-lineplot-following-y-value.html) and ggplot2 based [plot](https://stackoverflow.com/questions/35039846/change-line-color-depending-on-y-value-with-ggplot2?lq=1)


for annotation as well as flche et autre I think [BBC style](https://bbc.github.io/rcookbook/) will be required. 


[geom_area](https://www.r-bloggers.com/2017/09/how-to-make-a-line-chart-with-ggplot2/) to be fancy..but geom area coloring follwing a y axis



finally selest this [one](https://support.sisense.com/hc/en-us/community/posts/360038200234-Plotting-a-Gradient-Line-in-R)

[theme modification ggplot2](https://ggplot2.tidyverse.org/reference/theme.html)



I foun a tuto [here](https://msu.edu/course/zol/855/f09/workshop_111109.html) and the authors tuto [here](https://www.researchgate.net/publication/24309060_DNA_sequence_polymorphism_analysis_using_DnaSP/link/0912f50a8c3524b3c7000000/download) AND [HERE](http://www.ub.edu/dnasp/DnaSP6_Documentation_6.12.pdf)

i JUST SEE THAT IT IS POSSIBLE TO ASSIGN CODING REGIONS DEFIN THE TYPE OF DATA DNA/RNA..CHLOROPLAST OT MITO ...


This guy did a mistake [here](https://www.biostars.org/p/457878/)






convert chloae annotation gff to genbank

response [here](https://www.biostars.org/p/2492/)



